### PR TITLE
Add .spi.yml for Swift Package Index DocC support

### PR DESCRIPTION
**Pull request by yim-lee**

### CODE REVIEW

File: .spi.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+version: 1
+builder:
+  configs:
+    - documentation_targets: [SWIM]
```
Comment (ktoso) [inline .spi.yml:4]: Probably also ClusterMembership

**Review by ktoso — APPROVED** [review]

File: .spi.yml (hunk 1 above)
Comment (yim-lee) [reply .spi.yml:4]: But `ClusterMembership` doesn't have .docc dir?

Comment (ktoso) [reply .spi.yml:4]: Just it's type documentation should be useful enough already though, I don't think we _have_ to have a docc directory for type docs to be available 🤔

Comment (yim-lee) [reply .spi.yml:4]: ah ok

**Pull request merged**